exporters/prometheus/exporter.go

```diff
@@ -122,16 +122,18 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 		}
 	}
 
-	c.createTargetInfoOnce.Do(func() {
-		// Resource should be immutable, we don't need to compute again
-		targetInfo, err := c.createInfoMetric(targetInfoMetricName, targetInfoDescription, metrics.Resource)
+	c.mu.Lock()
```

Comment:
May be it will be easier to use lock for the whole Collect method not just critical sections?

Reply:
Maybe?  In that case it's important to document that `c.validateMetrics()` requires its caller to maintain a lock on the mutex.

Reply:
Decided to leave it as is because c.reader.Collect (L118) can take some time and it's actually safe.